command/issues/list/list.go

```diff
@@ -145,11 +143,39 @@ func (opts *IssuesListOptions) getIssuesData(ctx context.Context) (err error) {
 
 	// Fetch issues for a certain FileArg (filepath) passed by the user
 	// Example: `deepsource issues list api/hello.py`
-	if opts.FileArg != "" {
-		opts.issuesData, err = deepsource.GetIssuesForFile(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, opts.FileArg, opts.LimitArg)
-		if err != nil {
-			return err
+	if len(opts.FileArg) != 0 {
+		var fetchedIssues []issues.Issue
+		var filteredIssues []issues.Issue
+
+		for _, arg := range opts.FileArg {
+			// use regex to segregate directory and file names
+			var pathRegexp = regexp.MustCompile(`^(.*/)?(?:$|(.+?)(?:(\.[^.]*$)|$))`)
+			matched := pathRegexp.FindStringSubmatch(arg)
+
+			// if the argument contains a directory, fetch issues for the directory
+			if matched[1] != "" {
+				opts.issuesData, err = deepsource.GetIssues(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, opts.LimitArg)
+				if err != nil {
+					return err
+				}
+
+				filteredIssues, err = FilterIssuesByPath(arg, opts.issuesData)
+				if err != nil {
+					return err
+				}
+
+				fetchedIssues = append(fetchedIssues, filteredIssues...)
+			} else {
+				opts.issuesData, err = deepsource.GetIssuesForFile(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, arg, opts.LimitArg)
+				if err != nil {
+					return err
+				}
+				return nil
+			}
```

Comment:
Didn't completely understand the purpose of this block. I understand this is used when the argument is a file but we are overwriting the previous `opts.issuesData` with the newly fetched data. Should we `append` instead?

Reply:
Nice catch! Will fix it.